public/app/plugins/datasource/graphite/query_ctrl.ts

```diff
@@ -333,4 +346,126 @@ export class GraphiteQueryCtrl extends QueryCtrl {
       }
     }
   }
+
+  //////////////////////////////////
+  // Graphite seriesByTag support //
+  //////////////////////////////////
+
+  checkForSeriesByTag() {
```

Comment:
The GraphiteQueryCtrl is starting to be really big and do a little too much. I think it's time to break out a lot of the parsing and model building (And rendering back to string) to a separate model GraphiteQuery (I did this for InfluxDB a while back to clean up and simplify the query controller https://github.com/grafana/grafana/blob/master/public/app/plugins/datasource/influxdb/influx_query.ts  )